backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -66,14 +71,27 @@ private async void OnMissionRunCreated(object? sender, MissionRunCreatedEventArg
                 return;
             }
 
+            string missionRunIdToStart = missionRun.Id;
+
+            _scheduleLocalizationSemaphore.WaitOne();
+
+            string? localizationMissionRunId = null;
+            try { localizationMissionRunId = await LocalizationService.EnsureRobotIsCorrectlyLocalized(missionRun.Robot, missionRun); }
```

Comment:
Change to something like if (ShouldScheduleLocalizationMission)

Reply:
I'm not sure what we want with this comment? The function is in charge of scheduling the localization mission so the name ShouldScheduleLocalizationMission is not entirely correct. 

And we do use the exceptions to cancel the mission run if we fail localization?

Reply:
The comment was that it would perhaps be better to have the function return a boolean instead of using exceptions to signal if the check failed or not. Using exceptions is useful if we use the returned exceptions to do different behaviour depending on the type of exception, but in this case the check is true or false.